### PR TITLE
Tailor expressions in filter operation

### PR DESCRIPTION
**Pull request by dominiklohmann**
### :notebook_with_decorative_cover: Description

We ran into a `vast::die` call due to an unresolved field extractor when using the filter function with a field extractor. This most notably caused import filter like `event_type == "alert"` to fail.

### :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

Commit-by-commit.

### CODE REVIEW

File: libvast/src/table_slice.cpp [hunk 1]
```diff
@@ -706,9 +712,7 @@ filter(const table_slice& slice, const expression& expr, const ids& hints) {
   VAST_ASSERT(builder);
   auto flat_layout = flatten(slice.layout());
   auto check = [&](row_evaluator eval) {
-    if (expr == expression{})
-      return true;
-    return caf::visit(eval, expr);
+    return caf::visit(eval, *tailored_expr);
```
Comment (lava) [inline libvast/src/table_slice.cpp:715]: Isn't this inverting the logic for the case of an empty expression? As far as I understand we used to return everything and now we return nothing. (although the tests seem to pass, so I'm probably misreading this somehow)

**Review by lava — COMMENTED** [review]
The fix looks good, but from the description it sounds like we can very easily add the problematic query as a new test.

File: libvast/src/table_slice.cpp (hunk 1 above)
Comment (dominiklohmann) [reply libvast/src/table_slice.cpp:715]: Yep, and that's why the integration test is failing as well. Working on a fix rn.

**Review by lava — APPROVED** [review]
Looks good, thanks for adding the test!

**Pull request merged**